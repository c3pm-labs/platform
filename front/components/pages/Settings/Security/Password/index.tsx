import React from 'react';
import {
  makeStyles,
} from '@material-ui/core';
import * as yup from 'yup';
import { Formik, Form } from 'formik';

import PasswordInput from 'components/PasswordInput';

import SettingWrapper from '../../SettingWrapper';
import UpdateButton from '../../UpdateButton';

const useStyles = makeStyles((theme) => ({
  container: {
    display: 'flex',
    flex: 1,
    justifyContent: 'space-between',
    alignItems: 'flex-start',
    height: 'fit-content',
    [theme.breakpoints.down('xs')]: {
      justifyContent: 'center',
      alignItems: 'center',
      flex: 1,
      width: '100%',
    },
  },
  wrapperLabelAndInput: {
    display: 'flex',
    flex: 0.55,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingLeft: theme.spacing(5),
    minWidth: '200px',
    minHeight: '220px',
    [theme.breakpoints.down('xs')]: {
      paddingLeft: 0,
      flex: 1,
      justifyContent: 'center',
      width: '100%',
    },
  },
  containerInput: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    alignItems: 'center',
    flex: 0.8,
    minWidth: '220px',
    paddingRight: theme.spacing(2),
    [theme.breakpoints.down('xs')]: {
      paddingRight: 0,
      flex: 1,
    },
  },
  inivisiblelabel: {
    display: 'flex',
    flex: 0.2,
    paddingRight: theme.spacing(4),
    [theme.breakpoints.down('xs')]: {
      flex: 0,
      paddingRight: 0,
    },
  },
  styleForm: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    flex: 1,
    width: '100%',
    [theme.breakpoints.down('xs')]: {
      flexDirection: 'column',
    },
  },
}));

const passwordSchema = yup.object().shape({
  password: yup.string()
    .required('No password provided.')
    .min(8, 'Password is too short.'),
});

function Password(): JSX.Element {
  const classes = useStyles();

  return (
    <SettingWrapper
      title="Password :"
    >
      <Formik
        validationSchema={passwordSchema}
        initialValues={{ password: '' }}
        onSubmit={(e): void => (console.log('username', e))}
      >
        <div className={classes.container}>
          <Form className={classes.styleForm}>
            <div className={classes.wrapperLabelAndInput}>
              <div className={classes.inivisiblelabel} />
              <div className={classes.containerInput}>
                <PasswordInput
                  name="current password"
                  fullWidth
                  label="Current password"
                />
                <PasswordInput
                  name="password"
                  fullWidth
                  label="New password"
                />
                <PasswordInput
                  name="new password"
                  fullWidth
                  label="New password again"
                />
              </div>
            </div>
            <UpdateButton description="Update your password" onClick={(): void => (console.log('save'))} />
          </Form>
        </div>
      </Formik>
    </SettingWrapper>
  );
}

export default Password;
