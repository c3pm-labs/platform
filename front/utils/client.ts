import { ApolloClient, InMemoryCache, createHttpLink } from '@apollo/client'
import { setContext } from '@apollo/client/link/context'

const httpLink = createHttpLink({
  uri: process.env.GRAPHQL_URL,
  credentials: 'include',
})

const authLink = setContext((_, { headers }) => {
  return {
    headers: {
      ...headers,
    },
  }
})

const client = new ApolloClient({
  link: authLink.concat(httpLink),
  cache: new InMemoryCache(),
  ssrMode: typeof window === 'undefined',
})

export default client
