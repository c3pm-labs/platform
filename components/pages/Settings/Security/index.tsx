import React from 'react';
import {
  makeStyles,
} from '@material-ui/core';

import Password from './Password';

const useStyles = makeStyles((theme) => ({
  container: {
    display: 'flex',
    flex: 0.61,
    paddingLeft: theme.spacing(3),
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: theme.spacing(3),
  },
}));

function Security(): JSX.Element {
  const classes = useStyles();

  return (
    <div className={classes.container}>
      <Password />
    </div>
  );
}

export default Security;
