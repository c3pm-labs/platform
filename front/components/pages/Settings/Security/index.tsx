import React, { useState } from 'react';
import {
  makeStyles, Snackbar,
} from '@material-ui/core';
import { Form, Formik } from 'formik';
import { useMutation } from '@apollo/client';
import * as yup from 'yup';
import { Alert } from '@material-ui/lab';

import Button from '../../../Button';
import { useViewer } from '../../../../hooks/auth';
import { UPDATEPASSWORD } from '../../../../queries';
import PasswordInput from '../../../PasswordInput';

const useStyles = makeStyles((theme) => ({
  container: {
    display: 'flex',
    flex: 0.61,
    paddingLeft: theme.spacing(3),
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: theme.spacing(3),
  },
  form: {
    display: 'flex',
    flexDirection: 'column',
    width: '100%',
    marginTop: theme.spacing(4),
  },
  title: {
    color: theme.palette.primary.main,
    fontSize: '1.8em',
    fontWeight: 500,
    marginBottom: theme.spacing(4),
    marginTop: theme.spacing(2),
  },
  button: {
    marginTop: 10,
    width: 100,
    height: 40,
  },
  column: {
    display: 'flex',
    flexDirection: 'column',
    width: '100%',
    paddingBottom: theme.spacing(4),
    alignItems: 'flex-start',
    justifyContent: 'center',
  },
  input: {
    marginBottom: theme.spacing(3),
  },
}));

type SnackBarState = {
  type: 'success' | 'error';
  message?: string;
};

function Security(): JSX.Element {
  const classes = useStyles();
  const viewer = useViewer();
  const [isOpen, setIsOpen] = useState<boolean>(false);
  const [snackbar, setSnackbar] = useState<SnackBarState>({ type: 'success' });
  const [updatePassword] = useMutation(UPDATEPASSWORD, {
    onError: (error) => {
      setSnackbar({ type: 'error', message: error.message });
      setIsOpen(true);
    },
    onCompleted: () => {
      setSnackbar({ type: 'success', message: 'Password successfully updated!' });
      setIsOpen(true);
    },
  });

  const updateSchema = yup.object().shape({
    password: yup.string().required(),
    newPassword: yup.string().required().min(8),
  });

  if (!viewer) return <div>Loading...</div>;

  return (
    <div className={classes.container}>
      <Formik
        validationSchema={updateSchema}
        initialValues={{}}
        onSubmit={async (values): Promise<void> => {
          await updatePassword({
            variables: {
              id: viewer.id,
              ...values,
            },
          });
        }}
      >
        <Form className={classes.form}>
          <span className={classes.title}>Password :</span>
          <div className={classes.column}>
            <PasswordInput
              label="password"
              name="password"
              placeholder="current password"
              fullWidth
              className={classes.input}
              autoComplete="none"
            />
            <PasswordInput
              label="new password"
              name="newPassword"
              placeholder="new password"
              fullWidth
              autoComplete="none"
            />
            <Button type="submit" variant="contained" color="primary" className={classes.button}>
              Save
            </Button>
          </div>
        </Form>
      </Formik>
      <Snackbar open={isOpen} autoHideDuration={6000} onClose={() => setIsOpen(false)} anchorOrigin={{ vertical: 'top', horizontal: 'right' }}>
        <Alert onClose={() => setIsOpen(false)} severity={snackbar.type}>
          {snackbar.message}
        </Alert>
      </Snackbar>
    </div>
  );
}

export default Security;
