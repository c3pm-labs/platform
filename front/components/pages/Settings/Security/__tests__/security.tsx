import { MockedProvider } from '@apollo/client/testing';
import { render } from '@testing-library/react';

import Security from '../index';

// eslint-disable-next-line
const useViewer = jest.spyOn(require('hooks/auth'), 'useViewer');

test('Security', () => {
  const mocks = [];

  useViewer.mockImplementation(() => ({
    username: 'toto',
    email: 'tata',
  }));

  const { container } = render(
    <MockedProvider mocks={mocks} addTypename={false}>
      <Security />
    </MockedProvider>,
  );

  expect(container).toMatchSnapshot();
});
