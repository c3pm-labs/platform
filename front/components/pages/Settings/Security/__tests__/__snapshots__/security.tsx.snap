// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Security 1`] = `
<div>
  <div
    class="makeStyles-container-1"
  >
    <form
      action="#"
      class="makeStyles-form-2"
    >
      <span
        class="makeStyles-title-3"
      >
        Password :
      </span>
      <div
        class="makeStyles-column-5"
      >
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-input-6 MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
            data-shrink="false"
            for="password"
            id="password-label"
          >
            password
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <input
              aria-describedby="password-helper-text"
              aria-invalid="false"
              autocomplete="none"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              id="password"
              name="password"
              placeholder="current password"
              type="password"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-marginDense"
            >
              <button
                aria-label="toggle password visibility"
                class="MuiButtonBase-root MuiIconButton-root makeStyles-root-7 MuiIconButton-edgeEnd"
                data-testid="iconPasswordVisibility"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-11"
              >
                <span>
                  password
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            class="MuiFormHelperText-root makeStyles-root-8 MuiFormHelperText-contained MuiFormHelperText-marginDense"
            id="password-helper-text"
          >
            <span>
              ​
            </span>
          </p>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
            data-shrink="false"
            for="newPassword"
            id="newPassword-label"
          >
            new password
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <input
              aria-describedby="newPassword-helper-text"
              aria-invalid="false"
              autocomplete="none"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              id="newPassword"
              name="newPassword"
              placeholder="new password"
              type="password"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-marginDense"
            >
              <button
                aria-label="toggle password visibility"
                class="MuiButtonBase-root MuiIconButton-root makeStyles-root-7 MuiIconButton-edgeEnd"
                data-testid="iconPasswordVisibility"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-11"
              >
                <span>
                  new password
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            class="MuiFormHelperText-root makeStyles-root-8 MuiFormHelperText-contained MuiFormHelperText-marginDense"
            id="newPassword-helper-text"
          >
            <span>
              ​
            </span>
          </p>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-default-13 makeStyles-primary-14 makeStyles-primary-18 makeStyles-button-4 MuiButton-disableElevation"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Save
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </form>
  </div>
</div>
`;
